src/instructlab/data/generate.py

```diff
@@ -0,0 +1,218 @@
+@click.option(
+    "--model",
+    default=helpers.DEFAULT_MODEL,
+    show_default=True,
+    help="Name of the model used during generation.",
+)
+@click.option(
+    "--num-cpus",
+    type=click.INT,
+    help="Number of processes to use.",
+    default=helpers.DEFAULT_NUM_CPUS,
+    show_default=True,
+)
+@click.option(
+    "--chunk-word-count",
+    type=click.INT,
+    help="Number of words to chunk the document",
+    default=helpers.DEFAULT_CHUNK_WORD_COUNT,
+    show_default=True,
+)
+@click.option(
+    "--num-instructions",
+    type=click.INT,
+    help="Number of instructions to generate.",
+    default=helpers.DEFAULT_NUM_INSTRUCTIONS,
+    show_default=True,
+)
+@click.option(
+    "--taxonomy-path",
+    type=click.Path(),
+    default=helpers.DEFAULT_TAXONOMY_PATH,
+    show_default=True,
+    help=f"Path to {helpers.DEFAULT_TAXONOMY_REPO} clone or local file path.",
+)
+@click.option(
+    "--taxonomy-base",
+    default=helpers.DEFAULT_TAXONOMY_BASE,
+    show_default=True,
+    help="Base git-ref to use when generating new taxonomy.",
+)
+@click.option(
+    "--output-dir",
+    type=click.Path(),
+    default=helpers.DEFAULT_GENERATED_FILES_OUTPUT_DIR,
+    help="Path to output generated files.",
+)
+@click.option(
+    "--rouge-threshold",
+    type=click.FLOAT,
+    default=0.9,
+    show_default=True,
+    help="Threshold of (max) Rouge score to keep samples; 1.0 means accept all samples.",
+)
+@click.option(
+    "--quiet",
+    is_flag=True,
+    help="Suppress output of synthesized instructions.",
+)
+@click.option(
+    "--endpoint-url",
+    type=click.STRING,
+    help="Custom URL endpoint for OpenAI-compatible API. Defaults to the `ilab serve` endpoint.",
+)
+@click.option(
+    "--api-key",
+    type=click.STRING,
+    default=helpers.DEFAULT_API_KEY,  # Note: do not expose default API key
+    help="API key for API endpoint. [default: manage.DEFAULT_API_KEY]",
+)
+@click.option(
+    "--yaml-rules",
+    type=click.Path(),
+    default=helpers.DEFAULT_YAML_RULES,
+    show_default=True,
+    help="Rules file for YAML linting",
+)
+@click.option(
+    "--server-ctx-size",
+    type=click.INT,
+    default=helpers.MAX_CONTEXT_SIZE,
+    show_default=True,
+    help="The context size is the maximum number of tokens the server will consider.",
+)
+@click.option(
+    "--tls-insecure",
+    is_flag=True,
+    help="Disable TLS verification.",
+)
+@click.option(
+    "--tls-client-cert",
+    type=click.Path(),
+    default="",
+    show_default=True,
+    help="Path to the TLS client certificate to use.",
+)
+@click.option(
+    "--tls-client-key",
+    type=click.Path(),
+    default="",
+    show_default=True,
+    help="Path to the TLS client key to use.",
+)
+@click.option(
+    "--tls-client-passwd",
+    type=click.STRING,
+    default="",
+    help="TLS client certificate password.",
+)
+@click.option(
+    "--model-family",
+    help="Force model family to use when picking a generation template",
+)
```

Comment:
This seems like it's missing several fields

Reply:
which ones?

or is this comment now out of date?

Reply:
There's just `help` here - we should at least have a `type` I presume

Reply:
this is in the main branch as is